File: Module5/Case/RestFurama/src/main/java/thanh/code/models/AttachService.java
package thanh.code.models;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import com.fasterxml.jackson.annotation.JsonView;

import javax.annotation.Resource;
import javax.persistence.*;
import javax.validation.constraints.NotBlank;
import java.util.Set;

@Entity
public class AttachService {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int attachServiceId;
    @NotBlank
    private String attachServiceName;
    private double attachServiceCost;
    private int attachServiceUnit;
    private String attachServiceStatus;

    @OneToMany(mappedBy = "attachService", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    @JsonBackReference(value = "contractDetail")
    private Set<ContractDetail> contractDetail;

    public Set<ContractDetail> getContractDetail() {
        return contractDetail;
    }

    public void setContractDetail(Set<ContractDetail> contractDetail) {
        this.contractDetail = contractDetail;
    }

    public int getAttachServiceId() {
        return attachServiceId;
    }

    public void setAttachServiceId(int attachServiceId) {
        this.attachServiceId = attachServiceId;
    }

    public String getAttachServiceName() {
        return attachServiceName;
    }

    public void setAttachServiceName(String attachServiceName) {
        this.attachServiceName = attachServiceName;
    }

    public double getAttachServiceCost() {
        return attachServiceCost;
    }

    public void setAttachServiceCost(double attachServiceCost) {
        this.attachServiceCost = attachServiceCost;
    }

    public int getAttachServiceUnit() {
        return attachServiceUnit;
    }

    public void setAttachServiceUnit(int attachServiceUnit) {
        this.attachServiceUnit = attachServiceUnit;
    }

    public String getAttachServiceStatus() {
        return attachServiceStatus;
    }

    public void setAttachServiceStatus(String attachServiceStatus) {
        this.attachServiceStatus = attachServiceStatus;
    }

    public AttachService() {
    }
}

File: Module5/Case/FuramaNew/src/app/models/attach-service.ts
export class AttachService implements IAttachService{
  attachServiceCost: number;
  attachServiceId: number;
  attachServiceName: string;
  attachServiceStatus: string;
  attachServiceUnit: number;

  constructor(attachServiceCost: number, attachServiceId: number, attachServiceName: string, attachServiceStatus: string, attachServiceUnit: number) {
    this.attachServiceCost = attachServiceCost;
    this.attachServiceId = attachServiceId;
    this.attachServiceName = attachServiceName;
    this.attachServiceStatus = attachServiceStatus;
    this.attachServiceUnit = attachServiceUnit;
  }
}

export interface IAttachService {
  attachServiceId: number;
  attachServiceName: string;
  attachServiceCost: number;
  attachServiceUnit: number;
  attachServiceStatus: string;
}
